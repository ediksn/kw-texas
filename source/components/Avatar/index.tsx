import React, { useState } from 'react'
import { Image, ImageStyle, Text, TextStyle, View, ViewStyle } from 'react-native'
import DefaultAvatar from 'assets/images/default-avatar.png'
import { useSelector } from 'react-redux'
import { styles } from './styles'
import { useGetInitials } from '~/hooks'
import { ProfileInterface } from '~/interfaces/usrInterface'
import { RootState } from '~/store'

interface Props {
  uri?: any
  testID?: string
  avatarStyle?: ImageStyle
  avatarDefaultStyle?: ImageStyle
  initialsViewStyle?: ViewStyle
  initialsStyle?: TextStyle
  forceInitials?: boolean
}

const Avatar = ({
  uri,
  testID,
  avatarStyle,
  avatarDefaultStyle,
  initialsViewStyle,
  initialsStyle,
  forceInitials = false
}: Props) => {
  const profiles: ProfileInterface[] = useSelector((state: RootState) => state.usrProfile.profiles)
  const activeAccount: number = useSelector((state: RootState) => state.usrProfile.activeAccount)
  const currentProfile = (profiles && !!profiles.length && profiles[activeAccount]) || null
  const [imageError, setImageError] = useState(false)
  const uriOk = typeof uri !== 'undefined' && uri !== ''
  const imageOk = !imageError && uriOk

  return imageOk ? (
    <Image
      testID={testID}
      resizeMode='cover'
      source={{ uri }}
      style={[styles.avatar, avatarStyle]}
      onError={() => !imageError && setImageError(true)}
    />
  ) : !imageOk && forceInitials ? (
    <View style={[styles.initialsView, initialsViewStyle]}>
      <Text style={[styles.initials, initialsStyle]}>{useGetInitials(currentProfile?.name)}</Text>
    </View>
  ) : (
    <Image
      testID={testID}
      resizeMode='cover'
      source={DefaultAvatar}
      style={[styles.avatarDefault, avatarDefaultStyle]}
    />
  )
}

export default Avatar
