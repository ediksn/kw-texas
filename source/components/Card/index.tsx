import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import { styles } from './styles'
import mockPost from './__mocks__/mockPost'
import avatarMock from './__mocks__/avatarNinaRoyals.png'
import threeDotsMock from './__mocks__/threeDots.png'

const Card = () => {
  const { author, date } = mockPost

  const Header = () => (
    <View style={styles.header}>
      <Image style={styles.avatar} resizeMode='center' source={avatarMock} />
      <View style={styles.info}>
        <Text style={styles.name}>{author.name.toUpperCase()}</Text>
        <Text>{date}</Text>
      </View>
      <TouchableOpacity onPress={() => null}>
        <Image style={styles.threeDots} resizeMode='center' source={threeDotsMock} />
      </TouchableOpacity>
    </View>
  )

  return (
    <View style={styles.container}>
      <Header />
    </View>
  )
}

export default Card
