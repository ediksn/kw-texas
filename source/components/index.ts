/* eslint-disable import/no-cycle */
export { default as Avatar } from './Avatar'
export { default as Button } from './Button'
export { default as CheckBox } from './CheckBox'
export { default as Header } from './Header'
export { default as Input } from './Input'
export { default as Spinner } from './Spinner'
export { default as Video } from './NativeVideo'
export { default as VideoCard } from './VideoCard'
export { default as VideoList } from './VideoList'
export { default as HeaderTitleBox } from './HeaderTitleBox'
export { default as Post } from './Post'
export { default as Icon } from './Icon'
export { default as Dropdown } from './Dropdown'
export { default as Toast } from './Toast'
export { default as EventCard } from './EventCard'
export { default as EmptyList } from './EmptyList'
export { default as EventsCarrousel } from './Carrousel'
export { default as PostInfo } from './PostInfo'
export { default as PostMedia } from './PostMedia'
export { default as PostLinksForm } from './PostLinksForm'
export { default as TagsPeopleHeader } from './TagsPeopleHeader'
export { default as Tags } from './Tags'
