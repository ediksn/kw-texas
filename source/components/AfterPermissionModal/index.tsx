import React from 'react'
import { useTranslation } from 'react-i18next'
import { Linking, Platform, Text, View } from 'react-native'
import Modal from 'react-native-modal'
import Button from '../Button'
import { styles } from './styles'

interface Props {
  isVisible: boolean
  biometryType: string
  handleCloseModal: () => void
}
const AfterPermissionModal = ({ isVisible, biometryType, handleCloseModal }: Props) => {
  const { t } = useTranslation()

  const handleGoToSettings = () => {
    handleCloseModal()
    if (Platform.OS === 'ios') {
      Linking.openURL('app-settings://permissions/myapp')
    } else {
      Linking.openSettings()
    }
  }
  return (
    <Modal isVisible={isVisible} onBackdropPress={() => null}>
      <View style={styles.container}>
        <Text style={styles.titleText}>{t(`components_AfterPermissionModal_TurnOn${biometryType}`)}</Text>
        <Text style={styles.bodyText}>{t(`components_AfterPermissionModal_Opportunity${biometryType}`)}</Text>
        <View style={styles.buttonsContainer}>
          <Button
            viewStyle={styles.button}
            message={t('components_AfterPermissionModal_GoToSettings')}
            type='TEXT'
            onPress={handleGoToSettings}
          />
          <Button viewStyle={styles.button} message={t('general_text_Ok')} onPress={handleCloseModal} />
        </View>
      </View>
    </Modal>
  )
}

export default AfterPermissionModal
