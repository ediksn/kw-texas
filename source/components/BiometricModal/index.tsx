/** @format */

import React from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TouchableOpacity, View } from 'react-native'
import ReactNativeBiometrics from 'react-native-biometrics'

import { Icon } from '~/components'
import { theme } from '~/constants'

import { styles } from './styles'

interface Props {
  onAuth: any
  biometryType: string
  setBiometricPosition: React.Dispatch<React.SetStateAction<number>>
}

const BiometricModal = ({ onAuth, biometryType, setBiometricPosition }: Props) => {
  const { t } = useTranslation()

  const texts: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('components_BiometricModal_FaceID'),
    [ReactNativeBiometrics.Biometrics]: t('components_BiometricModal_Fingerprint'),
    [ReactNativeBiometrics.TouchID]: t('components_BiometricModal_Fingerprint')
  }
  const icons: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: 'faceid-icon',
    [ReactNativeBiometrics.TouchID]: 'fingerprint-icon',
    [ReactNativeBiometrics.Biometrics]: 'fingerprint-icon'
  }

  return (
    <View
      style={styles.container}
      onLayout={event => {
        setBiometricPosition(event.nativeEvent.layout.y)
      }}
    >
      <Text style={styles.text}>{texts[biometryType]}</Text>
      <TouchableOpacity onPress={onAuth}>
        <Icon size={40} name={icons[biometryType]} color={theme.activeColor} />
      </TouchableOpacity>
    </View>
  )
}

export default BiometricModal
