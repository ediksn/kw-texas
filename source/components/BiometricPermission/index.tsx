/** @format */

import React, { memo } from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TouchableOpacity, View } from 'react-native'
import ReactNativeBiometrics from 'react-native-biometrics'

import { ModalBottom } from '~/components/ModalBottom'

import { styles } from './styles'

interface Props {
  onYes: () => void
  onNo: () => void
  onRequestClose: () => void
  biometryType: string
  isVisible: boolean
}

const BiometricPermission = memo(({ onYes, onNo, onRequestClose, biometryType, isVisible }: Props) => {
  const { t } = useTranslation()

  const headers: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('Face ID'),
    [ReactNativeBiometrics.TouchID]: t('Fingerprint'),
    [ReactNativeBiometrics.Biometrics]: t('Fingerprint')
  }
  const bodies: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('Do you want to allow app to use Face ID to authenticate?'),
    [ReactNativeBiometrics.TouchID]: t('Do you want to allow app to use Fingerprint to authenticate?'),
    [ReactNativeBiometrics.Biometrics]: t('Do you want to allow app to use Fingerprint to authenticate?')
  }

  return (
    <ModalBottom visible={isVisible} onClose={onRequestClose}>
      <Text style={styles.header}>{headers[biometryType]}</Text>

      <Text style={styles.message}>{bodies[biometryType]}</Text>

      <View style={styles.buttons}>
        <TouchableOpacity onPress={onNo} style={[styles.button, styles.buttonNegative]}>
          <Text style={styles.textNegative}>{t('No')}</Text>
        </TouchableOpacity>

        <TouchableOpacity onPress={onYes} style={styles.button}>
          <Text style={styles.textPositive}>{t('Yes')}</Text>
        </TouchableOpacity>
      </View>
    </ModalBottom>
  )
})

export default BiometricPermission
