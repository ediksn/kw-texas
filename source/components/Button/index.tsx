import React, { memo } from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native'
import { moderateScale } from 'react-native-size-matters'
import { styles, stylesOfType } from './styles'
import { theme } from '~/constants'
import { Icon } from '~/components'

interface Props {
  testID?: string
  message?: string
  THEME?: any
  fontSize?: number
  type?: string
  viewStyle?: ViewStyle
  textStyle?: TextStyle
  disabled?: boolean
  onPress?: () => void
  icon?: any
}

const Button = ({
  testID,
  message = 'Ok',
  THEME = theme.buttons.primary,
  fontSize = moderateScale(theme.fonts.MEDIUM_SIZE),
  type = theme.buttons.types.CONTAINED,
  viewStyle,
  textStyle,
  disabled,
  onPress,
  icon = { name: '', size: 20, color: '#000000' }
}: Props) => {
  const { backgroundTypeStyle, textTypeStyle } = stylesOfType[type](THEME)

  const Message = () => {
    const { t } = useTranslation()

    return (
      <View style={[styles.messageView, icon && styles.messageWithIconView]}>
        {icon && (
          <View style={styles.icon}>
            <Icon name={icon.name} size={icon.size} color={icon.color} />
          </View>
        )}
        <Text style={[styles.messageText, { color: textStyle?.color }, textTypeStyle, { fontSize }]}>{t(message)}</Text>
      </View>
    )
  }

  return (
    <TouchableOpacity
      testID={testID}
      activeOpacity={0.6}
      disabled={disabled}
      style={[styles.containerView, viewStyle, backgroundTypeStyle]}
      onPress={onPress}
    >
      <Message />
    </TouchableOpacity>
  )
}

export default memo(Button)
