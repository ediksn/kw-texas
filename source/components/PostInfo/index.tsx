import React, { useState, useCallback, useEffect } from 'react'
import { View, Text, TouchableOpacity, ViewStyle, TouchableHighlight, Linking } from 'react-native'
import { useTranslation } from 'react-i18next'
import { moderateScale } from 'react-native-size-matters'
import { useDispatch } from 'react-redux'
import { styles } from './styles'
import { Button, PostMedia } from '~/components'
import { theme } from '~/constants'
import { Link, PostInterface } from '~/interfaces/postInterface'
import { useUnRichContent } from '~/hooks'
import { contentPost, buttonPost } from '../../constants/testIds'
import { homeActions } from '~/store/actions'

interface PostInfoProps {
  post: PostInterface
  commentsCount: number
  customContainerStyle?: ViewStyle
}

interface ButtonsProps {
  handleLikePost: () => void
  liked: boolean
}

const Buttons = ({ liked, handleLikePost }: ButtonsProps) => {
  const { t } = useTranslation()
  return (
    <View style={styles.buttons}>
      <Button
        testID={buttonPost}
        message={t('components_Post_Like')}
        type={theme.buttons.types.TEXT}
        icon={{
          name: liked ? 'like-filled-icon' : 'like-icon',
          size: moderateScale(20),
          color: theme.post.green
        }}
        fontSize={theme.fonts.SMALL_SIZE}
        viewStyle={styles.button}
        onPress={handleLikePost}
      />
      <Button
        testID={buttonPost}
        message={t('components_Post_Comment')}
        type={theme.buttons.types.TEXT}
        icon={{
          name: 'comment-icon',
          size: moderateScale(20),
          color: theme.post.green
        }}
        fontSize={theme.fonts.SMALL_SIZE}
        viewStyle={styles.button}
      />
      <Button
        testID={buttonPost}
        message={t('components_Post_Share')}
        type={theme.buttons.types.TEXT}
        icon={{
          name: 'share-icon',
          size: moderateScale(20),
          color: theme.post.green
        }}
        fontSize={theme.fonts.SMALL_SIZE}
        viewStyle={styles.button}
      />
    </View>
  )
}

const PostInfo = ({ post, commentsCount, customContainerStyle }: PostInfoProps) => {
  const { id, content, likesCount, userHasAlreadyLiked, detail } = post
  const shares = 0
  const MAX_LINES = 5
  const contentText = useUnRichContent(content)
  const [showMore, setShowMore] = useState(false)
  const [numberOfLines, setNumberOfLines] = useState<number | undefined>()
  const [hasShowLessMore, setHasShowLessMore] = useState(false)
  const [likes, setLikes] = useState(likesCount)
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const [liked, setLiked] = useState(userHasAlreadyLiked)

  const addMoreText = useCallback(
    (event: any) => {
      if (event.nativeEvent.lines.length > MAX_LINES && !showMore) {
        setHasShowLessMore(true)
      }
    },
    [showMore]
  )

  const handleOpenLink = async (url: string) => {
    const canOpen = await Linking.canOpenURL(url)
    if (canOpen) {
      await Linking.openURL(url)
    }
  }

  useEffect(() => {
    if (hasShowLessMore) {
      setNumberOfLines(showMore ? undefined : MAX_LINES)
    }
  }, [hasShowLessMore, showMore])

  const getLikesCommentsSharesText = (count: number, text: string) => {
    if (count > 0) {
      return `${count} ${count === 1 ? text.slice(0, text.length - 1) : text}`
    }

    return ''
  }

  const handleLikePost = () => {
    let likesUpdated
    if (!liked) {
      dispatch(homeActions.addLike(id))
      likesUpdated = likes + 1
    } else {
      dispatch(homeActions.removeLike(id))
      likesUpdated = likes - 1
    }
    setLiked(!liked)
    setLikes(likesUpdated)
  }

  const HorizontalLine = () => <View style={styles.horizontalLine} />

  return (
    <View style={styles.container}>
      <View>
        <View style={[styles.body, contentText ? customContainerStyle : {}]}>
          <View style={contentText ? styles.content : {}}>
            {contentText ? (
              <Text
                testID={contentPost}
                style={styles.contentText}
                numberOfLines={numberOfLines}
                onTextLayout={addMoreText}
                ellipsizeMode='clip'
              >
                {contentText}
              </Text>
            ) : null}
            {hasShowLessMore && (
              <TouchableOpacity onPress={() => setShowMore(!showMore)}>
                <Text style={styles.showMore}>
                  {showMore ? t('components_Post_ShowLess') : t('components_Post_ShowMore')}
                </Text>
              </TouchableOpacity>
            )}
          </View>
        </View>
        <PostMedia post={post} withTopMargin={false} />
        <View style={styles.body}>
          {detail?.links?.length && detail?.links?.length > 0 && (
            <View testID='links' style={styles.links}>
              {detail?.links?.map((link: Link, i: number) => (
                <TouchableHighlight
                  key={`key${0 + i}`}
                  onPress={() => handleOpenLink(link.url)}
                  underlayColor='transparent'
                  style={styles.linkWrapper}
                >
                  <Text style={styles.linkText}>{link.url}</Text>
                </TouchableHighlight>
              ))}
            </View>
          )}
          <View style={styles.infoNumbers}>
            <Text style={styles.infoNumber}>{getLikesCommentsSharesText(likes, t('components_Post_likes'))}</Text>
            <View style={styles.commentsSharesBox}>
              <Text style={[styles.infoNumber, shares > 0 ? styles.comments : null]}>
                {getLikesCommentsSharesText(commentsCount, t('components_Post_comments'))}
              </Text>
              <Text style={styles.infoNumber}>{getLikesCommentsSharesText(shares, t('components_Post_shares'))}</Text>
            </View>
          </View>
        </View>
      </View>
      <HorizontalLine />
      <Buttons liked={liked} handleLikePost={handleLikePost} />
      <HorizontalLine />
    </View>
  )
}

export default PostInfo
