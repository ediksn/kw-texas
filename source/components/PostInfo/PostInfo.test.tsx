import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import { postMockObject } from '../../../__mocks__/mockObjects'
import { PostInfo } from '~/components'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock('react-redux', () => ({
  useDispatch: () => ({})
}))

jest.mock(
  'rn-fetch-blob',
  () => {
    return {
      DocumentDir: () => {},
      ImageCache: {
        get: {
          clear: () => {}
        }
      },
      fs: {
        exists: jest.fn(),
        dirs: {
          MainBundleDir: () => {},
          CacheDir: () => {},
          DocumentDir: () => {}
        }
      }
    }
  },
  { virtual: true }
)

describe('Render Post Info Test', () => {
  let component: RenderAPI

  beforeEach(() => {
    component = render(<PostInfo post={postMockObject} commentsCount={1} customContainerStyle={{}} />)
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
  })
})
