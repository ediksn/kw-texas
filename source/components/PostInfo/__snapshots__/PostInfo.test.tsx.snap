// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Post Info Test Match with the snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 7.847058823529411,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 18.857142857142858,
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Object {
            "marginVertical": 7.847058823529411,
          }
        }
      >
        <Text
          ellipsizeMode="clip"
          onTextLayout={[Function]}
          style={
            Object {
              "color": "#3D424D",
              "flexWrap": "wrap",
              "fontFamily": "Mulish-Regular",
              "fontSize": 18.857142857142858,
            }
          }
          testID="content-post"
        >
          Post with images
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 18.857142857142858,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 25.142857142857142,
            "justifyContent": "space-between",
            "marginVertical": 19.61764705882353,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6C727E",
              "fontFamily": "Mulish-Regular",
              "fontSize": 18.857142857142858,
              "lineHeight": 25.142857142857142,
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#6C727E",
                  "fontFamily": "Mulish-Regular",
                  "fontSize": 18.857142857142858,
                  "lineHeight": 25.142857142857142,
                },
                null,
              ]
            }
          >
            1 components_Post_comment
          </Text>
          <Text
            style={
              Object {
                "color": "#6C727E",
                "fontFamily": "Mulish-Regular",
                "fontSize": 18.857142857142858,
                "lineHeight": 25.142857142857142,
              }
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#2b2b33",
        "height": 1.9617647058823529,
        "opacity": 0.1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 7.857142857142857,
          "borderWidth": 0,
          "elevation": 0,
          "height": 58.85294117647059,
          "justifyContent": "center",
          "margin": 7.857142857142857,
          "marginHorizontal": 8.571428571428571,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="button-post"
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginRight": 20.357142857142858,
              "textAlign": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 31.428571428571427,
                },
                Object {
                  "color": "#006C84",
                },
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#006C84",
                "fontFamily": "Mulish-Regular",
              },
              Object {
                "fontFamily": "Mulish-Bold",
              },
              undefined,
              Object {
                "fontSize": 18.857142857142858,
              },
            ]
          }
        >
          components_Post_Like
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 7.857142857142857,
          "borderWidth": 0,
          "elevation": 0,
          "height": 58.85294117647059,
          "justifyContent": "center",
          "margin": 7.857142857142857,
          "marginHorizontal": 8.571428571428571,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="button-post"
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginRight": 20.357142857142858,
              "textAlign": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 31.428571428571427,
                },
                Object {
                  "color": "#006C84",
                },
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#006C84",
                "fontFamily": "Mulish-Regular",
              },
              Object {
                "fontFamily": "Mulish-Bold",
              },
              undefined,
              Object {
                "fontSize": 18.857142857142858,
              },
            ]
          }
        >
          components_Post_Comment
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 7.857142857142857,
          "borderWidth": 0,
          "elevation": 0,
          "height": 58.85294117647059,
          "justifyContent": "center",
          "margin": 7.857142857142857,
          "marginHorizontal": 8.571428571428571,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="button-post"
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginRight": 20.357142857142858,
              "textAlign": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 31.428571428571427,
                },
                Object {
                  "color": "#006C84",
                },
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#006C84",
                "fontFamily": "Mulish-Regular",
              },
              Object {
                "fontFamily": "Mulish-Bold",
              },
              undefined,
              Object {
                "fontSize": 18.857142857142858,
              },
            ]
          }
        >
          components_Post_Share
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#2b2b33",
        "height": 1.9617647058823529,
        "opacity": 0.1,
      }
    }
  />
</View>
`;
